frontend/src/views/Room.jsx

```diff
@@ -256,6 +262,16 @@ function Room() {
 
       newRoom.on('ParticipantJoined', async (p) => {
         p.subscribe();
+        p.on('StartedSpeaking', () => {
+          const streamData = remoteStreamsRef.current.get(p.id);
+          streamData.speaking = true;
+          setRemoteStreamsRef(remoteStreamsRef.current);
+        });
+        p.on('StoppedSpeaking', () => {
+          const streamData = remoteStreamsRef.current.get(p.id);
+          streamData.speaking = false;
+          setRemoteStreamsRef(remoteStreamsRef.current);
+        });
```

Comment:
Can we encapsulate all of this in a function? I see that is duplicated when we join and then when a participant joins.